lib/Basic/Version.cpp

```diff
@@ -326,6 +327,9 @@ Optional<Version> Version::getEffectiveLanguageVersion() const {
   case 4:
     static_assert(SWIFT_VERSION_MAJOR == 4,
                   "getCurrentLanguageVersion is no longer correct here");
+    // Version '4' on its own implies '4.1'.
+    if (size() == 1)
+      return Version{4, 1};
     return Version::getCurrentLanguageVersion();
```

Comment:
I feel like this should at least *check* for 4.0 and 4.1.

Reply:
Check in what way?  The driver will disallow passing "4.0" and "4.1" explicitly.  The only thing the driver will allow is passing "4.2".  We could add an assert that, if there is a second component, that it is "2".  WDYT?

Reply:
Oh, right, the allowed-language-versions check. Okay, yes, that assert would be good enough for me.

Reply:
Ah, it shouldn't be 4.1, though. It needs to be higher than any 4.1 point updates we release, right? So 4.1.99 or something.

Reply:
I spoke wrong.  The allowed-language-versions check is defined in terms of `getEffectiveLanguageVersion`.